resolwe/flow/serializers/collection.py

```diff
@@ -18,6 +20,28 @@ class CollectionSerializer(ResolweBaseSerializer):
         allow_null=True,
         required=False,
     )
+    data_count = serializers.IntegerField()
+    status = serializers.SerializerMethodField()
+
+    def get_status(self, collection):
+        """Return status of the collection based on the status of data objects."""
+        if collection.data_count == 0:
+            return None
+
+        if collection.data_error_count:
+            return Data.STATUS_ERROR
+        if collection.data_uploading_count:
+            return Data.STATUS_UPLOADING
+        if collection.data_processing_count:
+            return Data.STATUS_PROCESSING
+        if collection.data_waiting_count:
+            return Data.STATUS_WAITING
+        if collection.data_resolving_count:
+            return Data.STATUS_RESOLVING
+        if collection.data_done_count == collection.data_count:
```

Comment:
You can move this if to the top and remove `== collection.data_count` part as the status will be overridden if any other data exists.